doc/bum_storm_control/bum_storm_control_hld.md

```diff
@@ -0,0 +1,399 @@
+      * [4.1 Enable storm control on physical interface](#41-enable-storm-control-on-phy)
+      * [4.2 Disable storm Control on physical interface](#42-disable-storm-control-on-phy)
+      * [4.3 Update storm Control on physical interface](#43-update-storm-control-on-phy)
+  * [5 Error Handling](#5-error-handling)
+  * [6 Serviceability and Debug](#6-serviceability-and-debug)
+  * [7 Warm Boot Support](#7-warm-boot-support)
+  * [8 Scalability](#8-scalability)
+  * [9 Unit Test](#9-unit-test) 
+     * [9.1 CLI Test Cases](#91-cli-test-cases)
+     * [9.2 Functional Test Cases](#92-functional-test-cases)
+     * [9.3 Scaling Test Cases](#93-scaling-test-cases)
+     * [9.4 Warm Boot Test Cases](#94-warm-boot-test-cases)
+     * [9.5 Negative Test Cases](#95-negative-test-cases)
+
+
+# List of Tables
+[Table 1: Abbreviations](#table-1-abbreviations)
+[Table 2: Storm Control SAI attributes](#table-2-storm-control-sai-attributes)
+
+
+# Revision
+| Rev |     Date    |       Author       | Change Description                |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 0.1 | 06/24/2019  |  Mohanarajan Selvaraj| Initial version                   |
+
+
+# About this Manual
+This document provides general information about Broadcast, Unknown-unicast and unknown-Multicast storm-control feature implementation in SONiC.
+# Scope
+This document describes the functionality and high level design of Broadcast, Unknown-unicast and unknown-Multicast storm-control feature in SONiC. 
+
+
+
+
+# Definition/Abbreviation
+### Table 1: Abbreviations
+| Term   | Meaning                                                |
+|--------|--------------------------------------------------------|
+| BUM   | Broadcast Unknown-unicast and unknown-Multicast         |
+
+
+# 1 Feature Overview
+A traffic storm occurs when packets flood the LAN, creating excessive traffic and degrading network performance. The type of traffic can be Broadcast, Unknown-unicast or unknown-Multicast (BUM). 
+The storm-control feature allows the user to limit the amount of BUM traffic admitted to the system. This can be achieved by configuring the type of storm (Broadcast or Unknown-unicast or unknown-Multicast) and the corresponding kilo bits per second (kbps) parameter on a given physical interface. Traffic that exceeds the configured rate will be dropped. 
+Unknown-multicast traffic consists of all multicast traffic which donot match any of the statically configured or dynamically learned multicast groups. 
+
+
+
+## 1.1 Requirements
+
+
+### 1.1.1 Functional Requirements
+ 1. Support configuration of Broadcast, Unknown-unicast and unknown-Multicast storm-control independently on physical interfaces.
+ 2. Support threshold rate configuration in kilo bits per second (kbps) in the range of 0 kbps to 100,000,000 kbps (100Gbps). 
+
+### 1.1.2 Configuration and Management Requirements
+This feature supports Click, Klish, REST, gNMI interfaces.
```

Comment:
Can we update as SONiC CLI (Click) and mgmt framework interfaces (CLI/REST/gNMI)?